extensions/security/runtime/src/main/java/io/quarkus/security/runtime/interceptor/SecurityHandler.java

```diff
@@ -0,0 +1,34 @@
+package io.quarkus.security.runtime.interceptor;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import javax.interceptor.InvocationContext;
+
+import io.quarkus.arc.runtime.InterceptorBindings;
+
+/**
+ * @author Michal Szynkiewicz, michal.l.szynkiewicz@gmail.com
+ *         <br>
+ *         Date: 05/10/2019
+ */
+@Singleton
+public class SecurityHandler {
+
+    private static final String HANDLER_NAME = SecurityHandler.class.getName();
+    private static final String EXECUTED = "executed";
+
+    @Inject
+    SecurityConstrainer constrainer;
+
+    public Object handle(InvocationContext ic) throws Exception {
+        if (alreadyHandled(ic)) {
```

Comment:
This might be a stupid question, but why is this check needed?

Reply:
it's only for preventing the check to be triggered multiple times, e.g. if one annotation is on the class, another on the method